doc/src/modules/combinatorics/fp_groups.rst

```diff
@@ -0,0 +1,185 @@
+Introduction
```

Comment:
This should probably be a next level header (underlined -----). The main header "FInitely Presented Groups" would come first (with ==== undeline). The lowest level headers, currently with ----. can then have e.g. `````.